server/cpho/views/indicator_data.py

```diff
@@ -396,7 +415,10 @@ def submission_status(self):
     def get_context_data(self, **kwargs):
         if self.dimension_type is None:
             predefined_dimension_types = DimensionType.objects.filter(
-                is_literal=False
+                is_literal=False,
+                code__in=relevant_dimension_types_for_period(
+                    self.indicator, self.period
+                ),
```

Comment:
Haven't tested this, but does this work considering we're comparing `DimensionType` records against `code` strings?